pkg/apis/application/v1alpha1/types.go

```diff
@@ -2278,7 +2278,7 @@ func (c *Cluster) RawRestConfig() *rest.Config {
 		} else {
 			config, err = clientcmd.BuildConfigFromFlags("", filepath.Join(os.Getenv("HOME"), ".kube", "config"))
 		}
-	} else if c.Server == KubernetesInternalAPIServerAddr && c.Config.Username == "" && c.Config.Password == "" && c.Config.BearerToken == "" {
```

Comment:
I think it will break existing in-clusters config with bearer tokens: Argo CD might incorrectly ignore configured in-cluster config if it has bearer token.

I bearer token is missing for EKS clusters etc. So I would suggest to add move checks into if instead of removing bearer token check:

```
} else if c.Server == KubernetesInternalAPIServerAddr && c.Config.Username == "" && c.Config.Password == "" && c.Config.BearerToken == "" && c.Config.AWSAuthConfig == nil && c.Config.ExecProviderConfig == nil 	{
```

Reply:
@alexmt the problem that we ran into was that for eks cluster certData and keyData were missing. And, because of that argocd-cli appended bearer token to cluster config (see logic below). And, hence cluster config was incorrectly selected to be TLSClientConfig instead of in-cluster config. 

https://github.com/argoproj/argo-cd/blob/cc4eea0d6951f1025c9ebb487374658186fa8984/cmd/util/cluster.go#L96-L98

Add more checks will not solve this problem #6233